### ISSUE TITLE
leave all layers

### ISSUE THREAD

**Issue by xcapegoat** [david-janssen/kmonad#37]
Hello,

Congrats on kmonad it's great. I am typing at normal speed while having space as a modifier, amazing really.

Two ideas, I thought it might be useful to:
- remove all layers from the stack at once and return to default
- add a configurable timeout to an added layer 

That is in case you wanted to leave nested layers without pressing LR binds in each regression. Is that possible btw?

Thanks.

**Comment by david-janssen** [comment]
Very happy to hear you're enjoying KMonad.

> That is in case you wanted to leave nested layers without pressing LR binds in each regression. Is that possible btw?

Not sure what you mean. Are you referring to a scenario where you add layers to the stack through a number of taps, and then want to return to an empty stack (base layer) with a single button?

> add a configurable timeout to an added layer

Not sure what you mean here either, could you give me a use-case example?

Cheerio

**Comment by xcapegoat** [comment]
> Not sure what you mean. Are you referring to a scenario where you add layers to the stack through a number of taps, and then want to return to an empty stack (base layer) with a single button?
 
Yea that's right. I see - LA-default does not add the default layer, it's already there. I got confused. So I just do LA-default. 

To be honest I can't think of a convincing use case for the second one. By a timeout I meant adding a time duration to a layer, after which it is removed automatically. It could be left to time out, or re-activated with certain keys. For example, a layer with a timeout of 1 where k and j emit up / down and restart the layer/timeout.

Most of the stuff you'd do with that could probably be done with scripts, like changing volume etc. Anyway my question was solved. Thanks again for your work on this, it's awesome. Cheers.
